src/pages/workspaces/workspace/applications/AppLauncher.js

```diff
@@ -15,13 +15,27 @@ import * as Utils from 'src/libs/utils'
 import { wrapWorkspace } from 'src/pages/workspaces/workspace/WorkspaceContainer'
 
 
+const getSparkInterfaceSource = (proxyUrl, sparkInterface) => {
+  console.assert(_.endsWith('/jupyter', proxyUrl), 'Unexpected ending for proxy URL')
+  const proxyUrlWithlastSegmentDropped = _.flow(_.split('/'), _.dropRight(1), _.join('/'))(proxyUrl)
+  return `${proxyUrlWithlastSegmentDropped}/${sparkInterface}`
+}
```

Comment:
Why is the string manipulation shenanigans, you might wonder 🤔  Turns out for Jupyter runtimes (both GCE and Dataproc), Leonardo returns a `proxyUrl` mapping to the `Jupyter` endpoint and not the Spark UI endpoints. Therefore we need to strip off `jupyter` at the end of the URL before adding the Spark UI-specific segment. I'm planning on filing a Leo ticket for us to consider returns a map of proxy URLs instead, now that we have several of them. With that, this string manipulation would go away and the `switchCase` below would be simpler.

Reply:
Might be good to capture this in a code comment